event/processor.go

```diff
@@ -81,53 +79,47 @@ func (p *eventProcessor) run(ctx context.Context, tail bool) error {
 		return nil
 	}
 
-	var err error
 	var collectors []types.ManagedObjectReference
 	for _, t := range p.tailers {
 		collectors = append(collectors, t.collector)
 	}
 
-	if len(p.tailers) > 1 {
-		// create and populate a ListView
-		viewMgr := view.NewManager(p.mgr.Client())
-		var listView *view.ListView
-		listView, err = viewMgr.CreateListView(ctx, collectors)
-		if err != nil {
-			return err
-		}
+	c := property.DefaultCollector(p.mgr.Client())
+	props := []string{"latestPage"}
 
-		count := 0
-		// Retrieve the property from the objects in the ListView
-		err = property.WaitForView(ctx, property.DefaultCollector(p.mgr.Client()), listView.Reference(), collectors[0], []string{latestPageProp}, func(c types.ManagedObjectReference, pc []types.PropertyChange) bool {
-			if err = p.process(c, pc); err != nil {
+	if len(collectors) == 1 {
+		// only one object to follow, don't bother creating a view
+		return property.Wait(ctx, c, collectors[0], props, func(pc []types.PropertyChange) bool {
+			if err := p.process(collectors[0], pc); err != nil {
```

Comment:
collectors is local to the run() method, and collectors[0] passes a copy of the types.ManagedObjectReference, so shouldn't be any issues there.